Libraries/Opc.Ua.Server/Diagnostics/CustomNodeManager.cs

```diff
@@ -1047,9 +1047,11 @@ public virtual void AddReferences(IDictionary<NodeId, IList<IReference>> referen
         /// This method validates any placeholder handle.
         /// </remarks>
         public virtual NodeMetadata GetNodeMetadata(
-            OperationContext context, 
-            object           targetHandle, 
-            BrowseResultMask resultMask)
+            OperationContext context,
+            object targetHandle,
+            BrowseResultMask resultMask,
+            Dictionary<NodeId, List<object>> uniqueNodesServiceAttributes = null,
+            bool permissionsOnly = false)
         {
```

Comment:
is there really no issue changing the signature, e.g. with compiled code? In the past I see there was a INodeManager2 added with additional functions. same could be done here with a GetNodeMetaData2 with more parameters in a INodeManager3?

Reply:
for compat it might also be etter to add the 5 parm function separately and not use the defaults. Maybe add it in INodeManager2 only?

Reply:
INodeManager2 now contains the 5 param function GetPermissionsMetadata which handles the optimizations and GetNodeMetadata is left as is.